systems/subscriber/subscriber-registry/pkg/db/subscriber_repo.go

```diff
@@ -0,0 +1,84 @@
+package db
+
+import (
+	"github.com/gofrs/uuid"
+	"github.com/ukama/ukama/systems/common/sql"
+	"gorm.io/gorm"
+)
+
+type SubscriberRepo interface {
+	Add(subscriber *Subscriber) error
+	Get(subscriberID uuid.UUID) (*Subscriber, error)
+	Delete(subscriberID uuid.UUID) error
+	Update(subscriberID uuid.UUID, sub Subscriber) (*Subscriber, error)
+	GetByNetwork(networkID uuid.UUID) ([]Subscriber, error)
+	ListSubscribers() ([]Subscriber, error)
+}
+
+type subscriberRepo struct {
+	Db sql.Db
+}
+
+func NewSubscriberRepo(db sql.Db) *subscriberRepo {
```

Comment:
Please, return the SubscriberRepo interface instead.

Reply:
Fixed.